sdk/mediaservices/azure-resourcemanager-mediaservices/src/test/java/com/azure/resourcemanager/mediaservices/ClientTests.java

```diff
@@ -0,0 +1,109 @@
+            "            }\n" +
+            "          ]\n" +
+            "        }\n" +
+            "      }\n" +
+            "    }";
+
+        Mockito.when(httpResponse.getStatusCode())
+            .thenReturn(200);
+        Mockito.when(httpResponse.getHeaders())
+            .thenReturn(new HttpHeaders());
+        Mockito.when(httpResponse.getBodyAsByteArray())
+            .thenReturn(Mono.just(response.getBytes(StandardCharsets.UTF_8)));
+
+        Mockito.when(httpClient.send(httpRequest.capture(), Mockito.any()))
+            .thenReturn(Mono.defer(() -> {
+                Mockito.when(httpResponse.getRequest())
+                    .thenReturn(httpRequest.getValue());
+                return Mono.just(httpResponse);
+            }));
+
+        MediaServicesManager manager = MediaServicesManager.configure()
+            .withHttpClient(httpClient)
+            .authenticate(
+                tokenRequestContext -> Mono.just(new AccessToken("this_is_a_token", OffsetDateTime.MAX)),
+                new AzureProfile("", "", AzureEnvironment.AZURE));
+
+        MediaService mediaservice = manager
+            .mediaservices()
+            .define("contososports")
+            .withRegion("South Central US")
+            .withExistingResourceGroup("contoso")
+            .withStorageAccounts(
+                Arrays
+                    .asList(
+                        new StorageAccount()
+                            .withId(
+                                "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/contoso/providers/Microsoft.Storage/storageAccounts/contososportsstore")
+                            .withType(StorageAccountType.PRIMARY)
+                            .withIdentity(
+                                new ResourceIdentity()
+                                    .withUserAssignedIdentity(
+                                        "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1")
+                                    .withUseSystemAssignedIdentity(false))))
+            .withStorageAuthentication(StorageAuthentication.MANAGED_IDENTITY)
+            .withEncryption(
+                new AccountEncryption()
+                    .withType(AccountEncryptionKeyType.CUSTOMER_KEY)
+                    .withKeyVaultProperties(
+                        new KeyVaultProperties().withKeyIdentifier("https://keyvault.vault.azure.net/keys/key1"))
+                    .withIdentity(
+                        new ResourceIdentity()
+                            .withUserAssignedIdentity(
+                                "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1")
+                            .withUseSystemAssignedIdentity(false)))
+            .create();
+
+        Assertions.assertNotNull(mediaservice.id());
+    }
```

Comment:
Ideally, we can add more assertion on the captured `httpRequest`, though it would be harder, as URL and query is composed in azure-core, not in codegen.